ibc-core/ics04-channel/src/handler/acknowledgement.rs

```diff
@@ -37,6 +37,7 @@ pub fn acknowledgement_packet_execute<ExecCtx>(
 ) -> Result<(), ContextError>
 where
     ExecCtx: ExecutionContext,
+    ClientStateMut<ExecCtx>: ClientStateExecution<ExecCtx::E>,
```

Comment:
Shouldn't this bound be part of `ExecutionContext` trait?

```rs
ClientStateMut<ExecCtx>: ClientStateExecution<ExecCtx::E>
```

Reply:
I see that it's already present

https://github.com/cosmos/ibc-rs/blob/4c4b72375592887844f513ba8fbc549a162b6626/ibc-core/ics24-host/src/context.rs#L150

Reply:
Yeah, exactly.
BTW, this is a part that I didn’t like first. Since it made us introduce such bounds everywhere. Though overall I think it's now considerably clean.

Reply:
It looks like my suggestion requires implied bounds (rust RFC 2089)
Check this comment - https://github.com/rust-lang/rust/issues/52662#issuecomment-473678551

Interestingly, Rust nightly can already do this via `associated_type_bounds`.

```rs
#![feature(associated_type_bounds)]

trait Color {
    type Channel;
}

trait CloneColor: Color<Channel: Clone> {}

fn foo<T: CloneColor>() {
    let f = T::Channel::clone;
}
````

Reply:
Thank you for investigating this.
Per our discussion, we're going away with this PR's implementation.